File: backend/app/schemas/documentation.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class DocumentationTypeItem(BaseModel):
    document_type: str
    title: str
    available: bool
    generated_by: str | None = None
    generated_at: datetime | None = None
    source_path: str | None = None


class DocumentationListResponse(BaseModel):
    types: list[DocumentationTypeItem]


class DocumentationResponse(BaseModel):
    document_type: str
    title: str
    content: str
    generated_by: str
    generated_at: datetime
    source_path: str | None = None

File: frontend/src/types/documentation.ts
export type DocumentGeneratedBy = "repository" | "ai";

export type DocumentType =
  | "repository_overview"
  | "architecture_overview"
  | "modules"
  | "classes"
  | "functions"
  | "branch_summary";

export interface DocumentationTypeItem {
  document_type: DocumentType;
  title: string;
  available: boolean;
  generated_by: DocumentGeneratedBy | null;
  generated_at: string | null;
  source_path: string | null;
}

export interface DocumentationListResponse {
  types: DocumentationTypeItem[];
}

export interface DocumentationResponse {
  document_type: DocumentType;
  title: string;
  content: string;
  generated_by: DocumentGeneratedBy;
  generated_at: string;
  source_path: string | null;
}

export const DOCUMENT_TYPE_LABELS: Record<DocumentType, string> = {
  repository_overview: "Repository Overview",
  architecture_overview: "Architecture Overview",
  modules: "Modules",
  classes: "Classes",
  functions: "Functions",
  branch_summary: "Branches",
};

export const DOCUMENT_CARD_ORDER: DocumentType[] = [
  "repository_overview",
  "architecture_overview",
  "modules",
  "classes",
  "functions",
  "branch_summary",
];
